service_crategen/src/commands/generate/codegen/mod.rs

```diff
@@ -441,7 +441,18 @@ fn generate_struct_fields<P: GenerateProtocol>(service: &Service,
             lines.push(format!("#[serde(rename=\"{}\")]", member_name));
 
             if let Some(shape_type) = service.shape_type_for_member(member) {
-                if shape_type == ShapeType::Blob {
+                if shape_type == ShapeType::Blob && !shape.required(member_name) {
```

Comment:
This might be more legible if the new `if` condition was introduced in a way that's decoupled from shape type.

You should be able to have multiple `#[serde]` attributs on a field, so you could have a single `if !shape.required(member_name)` which adds `#[serde(skip_serializing_if=\"Option::is_none\")` regardless of what the shape type is (so this logic could live outside of the block starting on line 443).

Reply:
That's a good point. For some reason I thought `#[serde]` attributes needed to be combined, and `master` had these two cases [in an `else if`](https://github.com/rusoto/rusoto/blob/master/service_crategen/src/commands/generate/codegen/mod.rs#L444-L454) configured, so I just expanded the matrix. I can instead hoist the existing optional block and be done.